### PR TITLE
Adding Vulnerability Assessment cmdlets on Server and Managed Instance

### PR DESCRIPTION
**Pull request by talhers**
## Description

These cmdlets are meant for setting Vulnerability Assessment settings on server and managed instance, and they are the same as the Vulnerability Assessment settings cmdlets on database and managed database.

Design review for Vulnerability Assessment cmdlets can be found at:
[https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/236](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/236)

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by talhers** [comment]
@yoavfr @yaakoviyun @jaredmoo

File: src/Sql/Sql/VulnerabilityAssessment/Cmdlet/SqlVulnerabilityAssessmentAtpCmdletBase.cs [hunk 1]
```diff
@@ -55,7 +55,7 @@ protected override M ApplyUserInputToModel(M model)
 
             // Verify that ATP is ON - if ATP is off the GetServerAdvancedThreatProtectionPolicy() will throw a not found exception
             var sqlAdvancedThreatProtectionAdapter = new SqlAdvancedThreatProtectionAdapter(DefaultProfile.DefaultContext);
-            if (GetResourceTypeVaAppliesTo() == ApplyToType.Database)
+            if (GetResourceTypeVaAppliesTo() == ApplyToType.Database || GetResourceTypeVaAppliesTo() == ApplyToType.Server)
```
[the remark this answers is not in the document]
Comment (talhers) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/SqlVulnerabilityAssessmentAtpCmdletBase.cs]: Yes, because the policy object returned is different, and the error message is different

File: src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/GetAzureSqlServerVulnerabilityAssessmentSettings.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        [ValidateNotNullOrEmpty]
+        public string ServerName { get; set; }
+
+        /// <summary>
+        /// Database resource
+        /// </summary>
+        [Parameter(ParameterSetName = GetByParentResourceParameterSet,
+            Mandatory = false,
+            ValueFromPipeline = true,
+            HelpMessage = "The database object to get Vulnerability Assessment settings for")]
```
[the remark this answers is not in the document]
Comment (talhers) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/GetAzureSqlServerVulnerabilityAssessmentSettings.cs]: Fixed

File: src/Sql/Sql.Test/ScenarioTests/VulnerabilityAssessmentMiTests.ps1 [hunk 3]
```diff
@@ -23,6 +23,211 @@ function Test-VulnerabilityAssessmentManagedInstanceWithAtpOffTest
 	Create-VulnerabilityAssessmentManagedInstanceTestEnvironment $testSuffix
 	$params = Get-SqlVulnerabilityAssessmentManagedInstanceTestEnvironmentParameters $testSuffix
 
+	try
+	{
+		$atpOffErrorMessage = "Managed Instance Advanced Threat Protection is not defined. Please run Enable-AzSqlInstanceAdvancedDataSecurity first."
```
[the remark this answers is not in the document]
Comment (talhers) [reply src/Sql/Sql.Test/ScenarioTests/VulnerabilityAssessmentMiTests.ps1]: Fixed

File: src/Sql/Sql/VulnerabilityAssessment/Services/BaseSqlVulnerabilityAssessmentAdapter.cs [hunk 4]
```diff
@@ -56,22 +56,34 @@ public BaseSqlVulnerabilityAssessmentAdapter(IAzureContext context)
...
+                    var vaManagedDbSettings = VulnerabilityAssessmentCommunicator.GetManagedDatabaseVulnerabilityAssessmentSettings(resourceGroup, serverName, databaseName);
+                    return ConvertSettingsToModel(resourceGroup, vaManagedDbSettings.RecurringScans, vaManagedDbSettings.StorageContainerPath);
+                case ApplyToType.Server:
+                    var vaServerSettings = VulnerabilityAssessmentCommunicator.GetServerVulnerabilityAssessmentSettings(resourceGroup, serverName);
+                    return ConvertSettingsToModel(resourceGroup, vaServerSettings.RecurringScans, vaServerSettings.StorageContainerPath);
+                case ApplyToType.ManagedInstance:
+                    var vaManagedInstanceSettings = VulnerabilityAssessmentCommunicator.GetManagedInstanceVulnerabilityAssessmentSettings(resourceGroup, serverName);
+                    return ConvertSettingsToModel(resourceGroup, vaManagedInstanceSettings.RecurringScans, vaManagedInstanceSettings.StorageContainerPath);
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(applyToType), applyToType, null);
```
[the remark this answers is not in the document]
Comment (talhers) [reply src/Sql/Sql/VulnerabilityAssessment/Services/BaseSqlVulnerabilityAssessmentAdapter.cs]: Fixed it to something more user-friendly

File: src/Sql/Sql/VulnerabilityAssessment/Cmdlet/SqlVulnerabilityAssessmentAtpCmdletBase.cs [hunk 5]
```diff
@@ -80,10 +80,13 @@ protected override M ApplyUserInputToModel(M model)
         /// </summary>
         public override void ExecuteCmdlet()
         {
-            if (GetDatabaseName().ToLower() == "master")
+            if (GetResourceTypeVaAppliesTo() == ApplyToType.Database || GetResourceTypeVaAppliesTo() == ApplyToType.ManagedDatabase)
             {
-                WriteWarning(Resources.VulnerabilityAssessmentOperationIsNotAvailableOnMasterDatabase);
-                return;
+                if (GetDatabaseName().ToLower() == "master")
```
[the remark this answers is not in the document]
Comment (talhers) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/SqlVulnerabilityAssessmentAtpCmdletBase.cs]: In managed instance we're not getting master at all. I changed the condition to check only in regular databases

**Comment by MiYanni** [comment]
@vladimir-shcherbakov Please reassign to someone else. Thanks.

**Comment by vladimir-shcherbakov** [comment]
@talhers Here Azure/azure-powershell-cmdlet-review-pr#78 I only found design review of the``` *SqlDatabaseVulnerabilityAssessment*```  cmdlets.
In the PR I found implementation of the cmdlts:
```
*SqlInstanceVulnerabilityAssessment*
*SqlServerVulnerabilityAssessment*
```
And .md files of the cmdlets
```
*SqlDatabaseVulnerabilityAssessment*
*SqlInstanceDatabaseVulnerabilityAssessment*
*SqlInstanceVulnerabilityAssessment*
```
How would you explain such a variety?

**Comment by yoavfr** [comment]
> @talhers Here [Azure/azure-powershell-cmdlet-review-pr#78](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/78) I only found design review of the` *SqlDatabaseVulnerabilityAssessment*` cmdlets.
> In the PR I found implementation of the cmdlts:
> 
> ```
> *SqlInstanceVulnerabilityAssessment*
> *SqlServerVulnerabilityAssessment*
> ```
> And .md files of the cmdlets
> 
> ```
> *SqlDatabaseVulnerabilityAssessment*
> *SqlInstanceDatabaseVulnerabilityAssessment*
> *SqlInstanceVulnerabilityAssessment*
> ```
> How would you explain such a variety?

@vladimir-shcherbakov 
The overall design for database vulnerability assessment applies with a minor change (no database parameter is provided) to the server level cmdlets. Everything else is identical.
The same is true for Managed Instance where everything is identical except for the name of the cmdlet and the reference inside to a differently named SDK (ManagedInstance instead of Server).
The help files have been altered for SqlDatabaseVulnerabilityAssessment since we now support server level policies (as opposed to database level) and this is the path that we recommend customers use.

**Comment by vladimir-shcherbakov** [comment]
@talhers 
You need to go through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) for every cmdlet with public API changes.

**Comment by vladimir-shcherbakov** [comment]
@talhers  Please fix Static Analysis issues: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell/10680/artifact/artifacts/BreakingChangeIssues.csv/*view*/

**Comment by vladimir-shcherbakov** [comment]
@talhers Since the [design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/236) has introduced only the  `AzSqlInstanceVulnerabilityAssessmentSettings` and `AzSqlServerVulnerabilityAssessmentSettings`  subset of cmdlets - to make it clearer please make sure no other cmdlets are modified in the PR.

**Comment by maddieclayton** [comment]
@talhers Ping on the comments above.

File: src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/ManagedInstance/ClearSqlManagedInstanceVulnerabilityAssessmentSettings.cs [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+namespace Microsoft.Azure.Commands.Sql.VulnerabilityAssessmentSettings.Cmdlet
+{
+    /// <summary>
+    /// Clears the Vulnerability Assessment on a specific managed instance.
+    /// </summary>
+    [Cmdlet("Clear", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlInstanceVulnerabilityAssessmentSettings", SupportsShouldProcess = true, DefaultParameterSetName = GetByNameParameterSet), OutputType(typeof(ManagedInstanceVulnerabilityAssessmentSettingsModel))]
+    public class ClearSqlManagedInstanceVulnerabilityAssessmentSettings : ClearSqlVulnerabilityAssessmentSettingsBase
+    {
+        /// <summary>
```
[the remark this answers is not in the document]
Comment (talhers) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/ManagedInstance/ClearSqlManagedInstanceVulnerabilityAssessmentSettings.cs]: managed instance cmdlet classes are named in the same manner, for example: GetAzureSqlManagedInstance

File: src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/ManagedInstance/GetAzureSqlManagedInstanceVulnerabilityAssessmentSettings.cs [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+using Microsoft.Azure.Commands.Sql.ManagedInstance.Model;
+
+namespace Microsoft.Azure.Commands.Sql.VulnerabilityAssessmentSettings.Cmdlet
+{
+    /// <summary>
+    /// Returns the Vulnerability Assessment settings of a specific managed instance.
+    /// </summary>
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlInstanceVulnerabilityAssessmentSettings", DefaultParameterSetName = GetByNameParameterSet), OutputType(typeof(ManagedInstanceVulnerabilityAssessmentSettingsModel))]
+    public class GetAzureSqlManagedInstanceVulnerabilityAssessmentSettings : GetAzureSqlVulnerabilityAssessmentSettingsBase
+    {
```
[the remark this answers is not in the document]
Comment (talhers) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/ManagedInstance/GetAzureSqlManagedInstanceVulnerabilityAssessmentSettings.cs]: managed instance cmdlet classes are named in the same manner, for example: GetAzureSqlManagedInstance

File: src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/ManagedInstance/UpdateAzureSqlManagedInstanceVulnerabilityAssessmentSettings.cs [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.VulnerabilityAssessmentSettings.Cmdlet
+{
+    /// <summary>
+    /// Updates the Vulnerability Assessment settings properties for a specific managed instance.
+    /// </summary>
+    [Cmdlet("Update", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlInstanceVulnerabilityAssessmentSettings", SupportsShouldProcess = true, DefaultParameterSetName = UpdateSettingsWithStorageAccountNameParameterSet), OutputType(typeof(ManagedInstanceVulnerabilityAssessmentSettingsModel))]
+    public class UpdateAzureSqlManagedInstanceVulnerabilityAssessmentSettings : UpdateAzureSqlVulnerabilityAssessmentSettingsBase
+    {
```
[the remark this answers is not in the document]
Comment (talhers) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/ManagedInstance/UpdateAzureSqlManagedInstanceVulnerabilityAssessmentSettings.cs]: Look at my previous answers, I follow the same naming convention as managed instance cmdlets

File: src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/GetAzureSqlServerVulnerabilityAssessmentSettings.cs [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+using Microsoft.Azure.Commands.Sql.Server.Model;
+
+namespace Microsoft.Azure.Commands.Sql.VulnerabilityAssessmentSettings.Cmdlet
+{
+    /// <summary>
+    /// Returns the Vulnerability Assessment settings of a specific server.
+    /// </summary>
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlServerVulnerabilityAssessmentSettings"), OutputType(typeof(ServerVulnerabilityAssessmentSettingsModel))]
+    public class GetAzureSqlServerVulnerabilityAssessmentSettings : GetAzureSqlVulnerabilityAssessmentSettingsBase
+    {
```
[the remark this answers is not in the document]
Comment (talhers) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/GetAzureSqlServerVulnerabilityAssessmentSettings.cs]: Look at my previous answers, I follow the same naming convention as managed instance cmdlets

File: src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/UpdateAzureSqlServerVulnerabilityAssessmentSettings.cs [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.VulnerabilityAssessmentSettings.Cmdlet
+{
+    /// <summary>
+    /// Updates the Vulnerability Assessment settings properties for a specific server.
+    /// </summary>
+    [Cmdlet("Update", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlServerVulnerabilityAssessmentSettings", SupportsShouldProcess = true, DefaultParameterSetName = UpdateSettingsWithStorageAccountNameParameterSet), OutputType(typeof(DatabaseVulnerabilityAssessmentSettingsModel))]
+    public class UpdateAzureSqlServerVulnerabilityAssessmentSettings : UpdateAzureSqlVulnerabilityAssessmentSettingsBase
+    {
```
[the remark this answers is not in the document]
Comment (talhers) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/UpdateAzureSqlServerVulnerabilityAssessmentSettings.cs]: Look at my previous answers, I follow the same naming convention as managed instance cmdlets

**Comment by talhers** [comment]
> @talhers Since the [design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/236) has introduced only the `AzSqlInstanceVulnerabilityAssessmentSettings` and `AzSqlServerVulnerabilityAssessmentSettings` subset of cmdlets - to make it clearer please make sure no other cmdlets are modified in the PR.

I removed all modifications to other cmdlets

**Comment by vladimir-shcherbakov** [comment]
@talhers Ping on the comments above.

File: src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/ManagedInstance/ClearSqlManagedInstanceVulnerabilityAssessmentSettings.cs [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.ManagedInstance.Model;
+
+namespace Microsoft.Azure.Commands.Sql.VulnerabilityAssessmentSettings.Cmdlet
+{
+    /// <summary>
+    /// Clears the Vulnerability Assessment on a specific managed instance.
+    /// </summary>
+    [Cmdlet("Clear", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlInstanceVulnerabilityAssessmentSettings", DefaultParameterSetName = ClearByNameParameterSet, SupportsShouldProcess = true), OutputType(typeof(ManagedInstanceVulnerabilityAssessmentSettingsModel))]
+    public class ClearSqlManagedInstanceVulnerabilityAssessmentSettings : ClearSqlVulnerabilityAssessmentSettingsBase
```
[the remark this answers is not in the document]
Comment (talhers) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/ManagedInstance/ClearSqlManagedInstanceVulnerabilityAssessmentSettings.cs]: It is handled in the base class - AzureRMCmdlet

File: src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/ManagedInstance/UpdateAzureSqlManagedInstanceVulnerabilityAssessmentSettings.cs [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Sql.VulnerabilityAssessmentSettings.Cmdlet
+{
+    /// <summary>
+    /// Updates the Vulnerability Assessment settings properties for a specific managed instance.
+    /// </summary>
+    [Cmdlet("Update", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlInstanceVulnerabilityAssessmentSettings", SupportsShouldProcess = true, DefaultParameterSetName = UpdateSettingsWithStorageAccountNameParameterSet), OutputType(typeof(ManagedInstanceVulnerabilityAssessmentSettingsModel))]
+    public class UpdateAzureSqlManagedInstanceVulnerabilityAssessmentSettings : UpdateAzureSqlVulnerabilityAssessmentSettingsBase
+    {
```
[the remark this answers is not in the document]
Comment (talhers) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/ManagedInstance/UpdateAzureSqlManagedInstanceVulnerabilityAssessmentSettings.cs]: It is handled in the base class - AzureRMCmdlet

File: src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/ClearSqlServerVulnerabilityAssessmentSettings.cs [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+using Microsoft.Azure.Commands.Sql.Server.Model;
+
+namespace Microsoft.Azure.Commands.Sql.VulnerabilityAssessmentSettings.Cmdlet
+{
+    /// <summary>
+    /// Clears the Vulnerability Assessment on a specific server.
+    /// </summary>
+    [Cmdlet("Clear", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlServerVulnerabilityAssessmentSettings", DefaultParameterSetName = ClearByNameParameterSet, SupportsShouldProcess = true), OutputType(typeof(ServerVulnerabilityAssessmentSettingsModel))]
+    public class ClearSqlServerVulnerabilityAssessmentSettings : ClearSqlVulnerabilityAssessmentSettingsBase
+    {
```
[the remark this answers is not in the document]
Comment (talhers) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/ClearSqlServerVulnerabilityAssessmentSettings.cs]: It is handled in the base class - AzureRMCmdlet

File: src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/UpdateAzureSqlServerVulnerabilityAssessmentSettings.cs [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Sql.VulnerabilityAssessmentSettings.Cmdlet
+{
+    /// <summary>
+    /// Updates the Vulnerability Assessment settings properties for a specific server.
+    /// </summary>
+    [Cmdlet("Update", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlServerVulnerabilityAssessmentSettings", SupportsShouldProcess = true, DefaultParameterSetName = UpdateSettingsWithStorageAccountNameParameterSet), OutputType(typeof(DatabaseVulnerabilityAssessmentSettingsModel))]
+    public class UpdateAzureSqlServerVulnerabilityAssessmentSettings : UpdateAzureSqlVulnerabilityAssessmentSettingsBase
+    {
```
[the remark this answers is not in the document]
Comment (talhers) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/UpdateAzureSqlServerVulnerabilityAssessmentSettings.cs]: It is handled in the base class - AzureRMCmdlet

File: src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/ClearSqlServerVulnerabilityAssessmentSettings.cs (hunk 13 above)
[the remark this answers is not in the document]
Comment (vladimir-shcherbakov) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/ClearSqlServerVulnerabilityAssessmentSettings.cs]: Could you please link me to the handler? I'm looking for something like this: 
```Powershell
[Cmdlet(VerbsCommon.Remove, "Foo", SupportsShouldProcess = true)]
public class RemoveFoo
{
    ...
    public override void ExecuteCmdlet()
    {
        if (ShouldProcess(target, actionMessage))
        {
            // make changes
        }
    }
}
```

**Comment by vladimir-shcherbakov** [comment]
@talhers Please resolve the merge conflicts and address the comment above.

File: src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/ClearSqlServerVulnerabilityAssessmentSettings.cs (hunk 13 above)
[the remark this answers is not in the document]
Comment (talhers) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/ClearSqlServerVulnerabilityAssessmentSettings.cs]: The handler is implemented in assembly Microsoft.Azure.PowerShell.Common
in class AzurePSCmdlet:

```powershell
protected virtual void ConfirmAction(string processMessage, string target, Action action)
{
	if (_qosEvent != null)
	{
		_qosEvent.PauseQoSTimer();
	}
	if (base.ShouldProcess(target, processMessage))
	{
		if (_qosEvent != null)
		{
			_qosEvent.ResumeQosTimer();
		}
		action();
	}
}
```


(class AzureRMCmdlet calls method ConfirmAction)

[the remark this answers is not in the document]
Comment (vladimir-shcherbakov) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/ClearSqlServerVulnerabilityAssessmentSettings.cs]: @talhers  The `ConfirmAction` is a [helper method](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/should-process-confirm-impact.md#additional-common-methods) that you may use to properly implement confirmation in the implementation of ProcessRecord()..

Only your cmdlet knows  which action should be executed in case of confirmation or `-Whatif`.

Please take a look at how the helper method is used in a random cmdlet from our repo: https://github.com/Azure/azure-powershell/blob/master/src/Network/Network/ApplicationGateway/RemoveAzureApplicationGatewayCommand.cs#L56

[the remark this answers is not in the document]
Comment (talhers) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/ClearSqlServerVulnerabilityAssessmentSettings.cs]: In Sql module, all cmdlet classes inherit from AzureSqlCmdletBase, and implement PersistChanges method.
PersistChanges is called from [AzureSqlCmdletBase.ExecuteCmdlet](https://github.com/Azure/azure-powershell/blob/master/src/Sql/Sql/Common/AzureSqlCmdletBase.cs#L119
) if ConfirmAction returns true.


For example, cmdlet [NewAzureSqlDatabase ](https://github.com/Azure/azure-powershell/blob/master/src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs)- uses "SupportsShouldProcess = true", but doesn't directly use ConfirmAction method, because it is used in ExecuteCmdlet in base class AzureSqlCmdletBase.

[the remark this answers is not in the document]
Comment (vladimir-shcherbakov) [reply src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/ClearSqlServerVulnerabilityAssessmentSettings.cs]: Now I see. Thanks.

File: src/Sql/Sql/help/Az.Sql.md [hunk 15]
```diff
@@ -167,9 +173,6 @@ Gets Advanced Data Security policy of a managed instance.
 ### [Get-AzSqlInstanceDatabase](Get-AzSqlInstanceDatabase.md)
 Returns information about Azure SQL Managed Instance database.
 
-### [Get-AzSqlInstanceDatabaseBackupShortTermRetentionPolicy](Get-AzSqlInstanceDatabaseBackupShortTermRetentionPolicy.md)
-Gets a backup short term retention policy.
-
```
Comment (vladimir-shcherbakov) [inline src/Sql/Sql/help/Az.Sql.md]: Why was this removed?

File: src/Sql/Sql/help/Clear-AzSqlInstanceVulnerabilityAssessmentSettings.md [hunk 16]
```diff
@@ -0,0 +1,156 @@
...
+```
+
+### Example 2 - Clear the vulnerability assessment settings from a managed instance object
+```powershell
+PS C:\> Get-AzSqlInstance `
+            -ResourceGroupName "ResourceGroup01" `
+            -Name "ManagedInstance01" `
+            | Clear-AzSqlInstanceVulnerabilityAssessmentSettings
+```
+
```
Comment (vladimir-shcherbakov) [inline src/Sql/Sql/help/Clear-AzSqlInstanceVulnerabilityAssessmentSettings.md]: Please add samples output

File: src/Sql/Sql/help/Clear-AzSqlServerVulnerabilityAssessmentSettings.md [hunk 17]
```diff
@@ -0,0 +1,156 @@
...
+```
+
+### Example 2 - Clear the vulnerability assessment settings from a server object
+```powershell
+PS C:\> Get-AzSqlServer `
+            -ResourceGroupName "ResourceGroup01" `
+            -ServerName "Server01" `
+            | Clear-AzSqlServerVulnerabilityAssessmentSettings
+```
+
```
Comment (vladimir-shcherbakov) [inline src/Sql/Sql/help/Clear-AzSqlServerVulnerabilityAssessmentSettings.md]: Please add samples output.

**Comment by vladimir-shcherbakov** [comment]
@talhers Ping on the comments above.

File: src/Sql/Sql/help/Az.Sql.md (hunk 15 above)
Comment (talhers) [reply src/Sql/Sql/help/Az.Sql.md]: I had a merge conflict where I added this line and it did not exist in the main repo. I guess it was removed by someone else, so I removed it too. This is not one of my cmdlets, so I will not add something owned by someone else

File: src/Sql/Sql/help/Clear-AzSqlInstanceVulnerabilityAssessmentSettings.md (hunk 16 above)
Comment (talhers) [reply src/Sql/Sql/help/Clear-AzSqlInstanceVulnerabilityAssessmentSettings.md]: Clear-AzSqlInstanceVulnerabilityAssessmentSettings doesn't return anything on success

File: src/Sql/Sql/help/Clear-AzSqlServerVulnerabilityAssessmentSettings.md (hunk 17 above)
Comment (talhers) [reply src/Sql/Sql/help/Clear-AzSqlServerVulnerabilityAssessmentSettings.md]: Clear-AzSqlServerVulnerabilityAssessmentSettings doesn't return anything on success

File: src/Sql/Sql/help/Clear-AzSqlInstanceVulnerabilityAssessmentSettings.md (hunk 16 above)
Comment (vladimir-shcherbakov) [reply src/Sql/Sql/help/Clear-AzSqlInstanceVulnerabilityAssessmentSettings.md]: if it doesn't - you may want to consider using `bool` for the output type of the [`Clear-AzSqlInstanceVulnerabilityAssessmentSettings`](https://github.com/Azure/azure-powershell/pull/8413/files?file-filters%5B%5D=.cs#diff-632902af0776c8292c91cb56a243ae9dR26) and [`Clear-AzSqlServerVulnerabilityAssessmentSettings`](https://github.com/Azure/azure-powershell/pull/8413/files?file-filters%5B%5D=.cs#diff-10494e091851c462e9f1cd916383f4abR26) cmdlets and add `-PassThru` parameter.

File: src/Sql/Sql/help/Clear-AzSqlServerVulnerabilityAssessmentSettings.md (hunk 17 above)
Comment (vladimir-shcherbakov) [reply src/Sql/Sql/help/Clear-AzSqlServerVulnerabilityAssessmentSettings.md]: the same comment.

**Comment by vladimir-shcherbakov** [comment]
@talhers  Ping on the comments above.

File: src/Sql/Sql/help/Clear-AzSqlInstanceVulnerabilityAssessmentSettings.md (hunk 16 above)
Comment (talhers) [reply src/Sql/Sql/help/Clear-AzSqlInstanceVulnerabilityAssessmentSettings.md]: These cmdlets are consistent with our other cmdlets (for databases and managed databases) so we will keep it as it is for now. We might change that in the future for all our cmdlets.

**Comment by talhers** [comment]
Note on latest commit:

We discovered a bug where users with VA policy defined only at server level were not able to perform actions with our cmdlets (such as scan, baseline, export etc) because all VA cmdlets were checking that VA policy is defined at database level. This check is not done properly and is also redundant because it is performed at the backend.

Therefore I removed these checks from our cmdlets (remove SqlVulnerabilityAssessmentOperationCmdletBase class and remove method ApplyUserInputToModel from SqlVulnerabilityAssessmentAtpCmdletBase class).

These checks are performed in the backend so the cmdlets' behavior will remain the same.
I have updated the tests and added a case for checking this both on server and on managed instance.

File: src/Sql/Sql/help/Clear-AzSqlInstanceVulnerabilityAssessmentSettings.md (hunk 16 above)
Comment (vladimir-shcherbakov) [reply src/Sql/Sql/help/Clear-AzSqlInstanceVulnerabilityAssessmentSettings.md]: It's not clear what you mean by the **consistency**  
* if the other cmdlets don't use their output types - you need to deprecate them following the document: https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-attribute-help.md#cmdletoutputbreakingchangeattribute. 
* else - please use the output types to return anything on success and update corresponding help files samples.

**Comment by yoavfr** [comment]
Looks good

**Comment by vladimir-shcherbakov** [comment]
signing: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/psnetsign-esrp/250/

File: src/Sql/Sql/help/Update-AzSqlDatabaseVulnerabilityAssessmentSettings.md [hunk 18]
```diff
@@ -223,7 +223,7 @@ Parameter Sets: UpdateByInputObjectParameterSet
 Aliases:
 
 Required: True
-Position: Named
+Position: 0
 Default value: None
```
Comment (vladimir-shcherbakov) [inline src/Sql/Sql/help/Update-AzSqlDatabaseVulnerabilityAssessmentSettings.md]: I not sure why you touched this file but it's a breaking change.

File: src/Sql/Sql/help/Update-AzSqlInstanceDatabaseVulnerabilityAssessmentSettings.md [hunk 19]
```diff
@@ -223,7 +223,7 @@ Parameter Sets: UpdateByInputObjectParameterSet
 Aliases:
 
 Required: True
-Position: Named
+Position: 0
```
Comment (vladimir-shcherbakov) [inline src/Sql/Sql/help/Update-AzSqlInstanceDatabaseVulnerabilityAssessmentSettings.md]: I not sure why you touched this file but it's a breaking change.

File: src/Sql/Sql/help/Update-AzSqlDatabaseVulnerabilityAssessmentSettings.md (hunk 18 above)
Comment (talhers) [reply src/Sql/Sql/help/Update-AzSqlDatabaseVulnerabilityAssessmentSettings.md]: Fixed

File: src/Sql/Sql/help/Update-AzSqlInstanceDatabaseVulnerabilityAssessmentSettings.md (hunk 19 above)
Comment (talhers) [reply src/Sql/Sql/help/Update-AzSqlInstanceDatabaseVulnerabilityAssessmentSettings.md]: Fix
